feat: create and send event about what happened

Godley opened the pull request:
Persists chaos monkey deletion to event layer for easy assessing of whether the chaos monkey caused the problem or something else.
Still to test this locally, but in theory this should be ok.

Commit:
feat: create and send event about what happened

Persists chaos monkey deletion to event layer for easy assessing of whether
the chaos monkey caused the problem or something else

mikebryant commented on chaos.py:
```diff
@@ -25,5 +25,15 @@
+            uid=pod.metadata.uid
+        ),
+        message="Pod deleted by chaos monkey",
+        type="Warning"
```
nitpick: Add trailing comma?

mikebryant commented on chaos.py:
```diff
@@ -25,5 +25,15 @@
+            kind="Pod",
+            name=pod.metadata.name,
+            namespace=pod.metadata.namespace,
+            uid=pod.metadata.uid
```
nitpick: trailing comma

mikebryant reviewed (changes requested):
Looks good.
We should definitely test it locally before merge though

Commit:
fix(event-name): add event name

Why: api will return unprocessable entity
otherwise

Godley commented:
@mikebryant tested on minikube, just one problem with the event not having a name then it seems to be working fine.

Commit:
feat: increment count if there already

- if we deleted the pod previously, get the old event and increment the count
- error handling to figure out whether to increment or create

Godley commented:
I'm getting "Event in version \"v1\" cannot be handled as a Event: only encoded map or array can be decoded into a struct" when I try to create the event from the exception handler. Completely non-plussed as to why.

Commit:
fix: event source is correct object type

also added timestamps + reraising

Godley commented:
@mikebryant should be all good now

mikebryant commented:
LGTM
Can we squash that last fix commit?

mikebryant commented:
and also the style commits etc

Godley commented:
Sure, I'm not sure how to merge commits into each other though?
